release/internal/version/version.go

```diff
@@ -36,25 +36,27 @@ type Data interface {
 	ReleaseBranch(releaseBranchPrefix string) string
 }
 
-func NewVersionData(calico Version, operator string) Data {
+func NewVersionData(calico Version, operator, releaseName string) Data {
 	return &CalicoVersionData{
-		calico:   calico,
-		operator: operator,
+		calico:      calico,
+		operator:    operator,
+		releaseName: releaseName,
 	}
 }
 
 // CalicoVersionData provides version data for a Calico release.
 type CalicoVersionData struct {
-	calico   Version
-	operator string
+	calico      Version
+	operator    string
+	releaseName string
 }
 
 func (v *CalicoVersionData) ProductVersion() string {
 	return v.calico.FormattedString()
 }
 
 func (v *CalicoVersionData) OperatorVersion() string {
-	return v.operator
+	return fmt.Sprintf("%s-%s", v.operator, v.releaseName)
```

Comment:
Couldn't the operator version just include the productVer and that would be unique?

Is this function called for real releases at all? This format would be wrong for a real release (I don't think it is... but wanted to check)

Reply:
the previous process did use the release name. Although using the product version is much better than an arbitrary release name.

This function/ struct is only used for hashreleases.